packaging/greenboot/microshift-greenboot-required.sh

```diff
@@ -0,0 +1,209 @@
+#
+# args: None
+# env1: 'CHECK_PODS_NS' environment variable for the namespace to check
+# env2: 'CHECK_PODS_CT' environment variable for the pod count to check
+# return: 0 if the expected number of pods are ready, or 1 otherwise
+function namespace_pods_ready() {
+    local ns=${CHECK_PODS_NS}
+    local ct=${CHECK_PODS_CT}
+
+    local status=$(oc get pods ${OCGET_OPT} -n ${ns} -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}' 2>/dev/null)
+    local tcount=$(echo $status | grep -o True  | wc -l)
+    local fcount=$(echo $status | grep -o False | wc -l)
+
+    # Terminate the script in case more pods are ready than expected - nothing to wait for
+    if [ "${tcount}" -gt "${ct}" ] ; then
+        log "The number of ready pods in the '${ns}' namespace is greater than the expected '${ct}' count. Terminating..."
+        kill -TERM ${SCRIPT_PID}
+    fi
+    # Exit with error if any pods are not ready yet
+    [ "${fcount}" -gt 0 ] && return 1
+    # Check the ready pod count    
+    [ "${tcount}" -eq "${ct}" ] && return 0
+    return 1
+}
+
+# Check if MicroShift pods in a given namespace started and verify they are not restarting by sampling
+# the pod restart count 10 times every 5 seconds and comparing the current sample with the previous one.
+# The pods are considered restarting if the number of 'pod-restarting' samples is greater than the
+# number of 'pod-not-restarting' ones.
+#
+# arg1: Name of the namespace to check
+# return: 0 if pods are not restarting, or 1 otherwise
+function namespace_pods_not_restarting() {
+    local ns=$1
+    local restarts=0
+
+    local count1=$(oc get pods ${OCGET_OPT} -n ${ns} -o 'jsonpath={..status.containerStatuses[].restartCount}' 2>/dev/null)
+    for i in $(seq 10) ; do
+        sleep 5
+        local countS=$(oc get pods ${OCGET_OPT} -n ${ns} -o 'jsonpath={..status.containerStatuses[].started}' 2>/dev/null | grep -vc false)
+        local count2=$(oc get pods ${OCGET_OPT} -n ${ns} -o 'jsonpath={..status.containerStatuses[].restartCount}' 2>/dev/null)
+
+        # If pods started, a restart is detected by comparing the count string between the checks.
+        # The number of pod restarts is incremented when a restart is detected, or decremented otherwise.
+        if [ "${countS}" -ne 0 ] && [ "${count1}" = "${count2}" ] ; then
+            restarts=$(( restarts - 1 ))
+        else
+            restarts=$(( restarts + 1 ))
+            count1=${count2}
+        fi
+    done
+
+    [ "${restarts}" -lt 0 ] && return 0
+    return 1
+}
+
+#
+# Main
+#
+log "STARTED"
+
+# Dump GRUB boot and Greenboot variables if any
+grub_vars=$(grub2-editenv - list | grep ^boot_ || true)
+boot_vars=$(set | egrep '^GREENBOOT_|^MICROSHIFT_' || true)
+
+[ -z "${grub_vars}" ] && grub_vars=None
+[ -z "${boot_vars}" ] && boot_vars=None
+
+log "GRUB boot variables:\n${grub_vars}"
+log "Greenboot variables:\n${boot_vars}"
+
+# Exit if the ostree filesystem is not used
```

Comment:
In general, greenboot RPM can be installed on a non-ostree file system as well. It would run the checks, without the ability to rollback. I'm removing this check so that the script can be used regardless of the ostree presence.